main.go

```diff
@@ -124,8 +125,16 @@ func main() {
 
 	// Start HTTP server
 	address := net.JoinHostPort(cfg.Exporter.Host, strconv.Itoa(cfg.Exporter.Port))
+	srv := &http.Server{Addr: address}
+	go func() {
+		<-ctx.Done()
+		if err := srv.Shutdown(context.Background()); err != nil {
+			logger.Error("error stopping HTTP server", zap.Error(err))
+			os.Exit(1)
+		}
+	}()
```

Comment:
Ah yes, I saw that goroutine, but decided to not use for two reasons:
1. I will need to move it further down below, after the creation of the `srv` instance.
2. The goroutine currently has a single `select`, which means that it will kick either if the `c` channel gets a signal OR the `ctx` context is done somehow else. The solution would be to place the `select` inside an infinite loop that breaks in the `ctx.Done()` case.

Therefore I decided to make an additional goroutine to separate concerns. If you still want to use a single goroutine, I can work out the two above points. Let me know!

Reply:
@weeco I realised that the signal handling code can be greatly simplified by using a signal-stoppable context created natively by the `signal` package (since Go 1.16). This removes the need for a handling goroutine and the code is cleaner.

Let me know if you need anything else.

Reply:
Makes sense, thank you